src/Libraries/Microsoft.Extensions.AI.Abstractions/Contents/HostedFileContent.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using Microsoft.Shared.Diagnostics;
+
+namespace Microsoft.Extensions.AI;
+
+/// <summary>
+/// Represents a file that is hosted by the AI service.
+/// </summary>
+/// <remarks>
+/// Unlike <see cref="DataContent"/> which contains the data for a file or blob, this class represents a file that is hosted
+/// by the AI service and referenced by an identifier. Such identifiers are specific to the provider.
+/// </remarks>
+[DebuggerDisplay("FileId = {FileId}")]
+public sealed class HostedFileContent : AIContent
+{
+    private string _fileId;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="HostedFileContent"/> class.
+    /// </summary>
+    /// <param name="fileId">The ID of the hosted file.</param>
+    /// <exception cref="ArgumentNullException"><paramref name="fileId"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="fileId"/> is empty or composed entirely of whitespace.</exception>
+    public HostedFileContent(string fileId)
+    {
+        _fileId = Throw.IfNullOrWhitespace(fileId);
+    }
+
+    /// <summary>
+    /// Gets or sets the ID of the hosted file.
+    /// </summary>
+    /// <exception cref="ArgumentNullException"><paramref name="value"/> is <see langword="null"/>.</exception>
+    /// <exception cref="ArgumentException"><paramref name="value"/> is empty or composed entirely of whitespace.</exception>
+    public string FileId
```

Comment:
This seems a useful holder, but I would expect to do anything useful with it we might also need abstractions for creation, access, removal.  Perhaps a "metadata" concept as well.

Reply:
@ericstj This is a valid point, currently the API tools have only exposed the file identifier and indirectly what is the type "text" or "image", we might consider adding extra properties if we get multiple API's providing more data about the file during the tool config/response, for now the path for those would be using the `AdditionalProperties` dictionary.